src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [chosenPerson, setChosenPerson] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const handleQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value);
+    applyQuery(e.target.value);
+    setChosenPerson('');
+  };
+
+  const onSelected = (personName: string) => {
+    setQuery(personName);
+    setChosenPerson(personName);
+  };
+
+  const createTitle = (personName: string) => {
+    const person = peopleFromServer.find(
+      findingPerson => findingPerson.name === personName,
+    );
+
+    if (!person) {
+      return personName;
+    }
+
+    return `${personName} (${person.born} - ${person.died})`;
+  };
+
+  const suggestions = useMemo(() => {
+    return peopleFromServer.filter(folk => folk.name.includes(appliedQuery));
+  }, [appliedQuery]);
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {chosenPerson ? createTitle(chosenPerson) : 'No selected person'}
```

Comment:
You are interacting with the DOM directly using 'document.getElementById'. In React, it's recommended to use the React way of doing things as much as possible. In this case, you can use state and props to handle the visibility of elements.